src/neon/neon_planner.rs

```diff
@@ -0,0 +1,836 @@
+                } else if id_t == id_f64 {
+                    Arc::new(NeonF64Butterfly8::new(direction)) as Arc<dyn Fft<T>>
+                } else {
+                    panic!("Not f32 or f64");
+                }
+            }
+            Recipe::Butterfly9 => {
+                if id_t == id_f32 {
+                    Arc::new(NeonF32Butterfly9::new(direction)) as Arc<dyn Fft<T>>
+                } else if id_t == id_f64 {
+                    Arc::new(NeonF64Butterfly9::new(direction)) as Arc<dyn Fft<T>>
+                } else {
+                    panic!("Not f32 or f64");
+                }
+            }
+            Recipe::Butterfly10 => {
+                if id_t == id_f32 {
+                    Arc::new(NeonF32Butterfly10::new(direction)) as Arc<dyn Fft<T>>
+                } else if id_t == id_f64 {
+                    Arc::new(NeonF64Butterfly10::new(direction)) as Arc<dyn Fft<T>>
+                } else {
+                    panic!("Not f32 or f64");
+                }
+            }
+            Recipe::Butterfly11 => {
+                if id_t == id_f32 {
+                    Arc::new(NeonF32Butterfly11::new(direction)) as Arc<dyn Fft<T>>
+                } else if id_t == id_f64 {
+                    Arc::new(NeonF64Butterfly11::new(direction)) as Arc<dyn Fft<T>>
+                } else {
+                    panic!("Not f32 or f64");
+                }
+            }
+            Recipe::Butterfly12 => {
+                if id_t == id_f32 {
+                    Arc::new(NeonF32Butterfly12::new(direction)) as Arc<dyn Fft<T>>
+                } else if id_t == id_f64 {
+                    Arc::new(NeonF64Butterfly12::new(direction)) as Arc<dyn Fft<T>>
+                } else {
+                    panic!("Not f32 or f64");
+                }
+            }
+            Recipe::Butterfly13 => {
+                if id_t == id_f32 {
+                    Arc::new(NeonF32Butterfly13::new(direction)) as Arc<dyn Fft<T>>
+                } else if id_t == id_f64 {
+                    Arc::new(NeonF64Butterfly13::new(direction)) as Arc<dyn Fft<T>>
+                } else {
+                    panic!("Not f32 or f64");
+                }
+            }
+            Recipe::Butterfly15 => {
+                if id_t == id_f32 {
+                    Arc::new(NeonF32Butterfly15::new(direction)) as Arc<dyn Fft<T>>
+                } else if id_t == id_f64 {
+                    Arc::new(NeonF64Butterfly15::new(direction)) as Arc<dyn Fft<T>>
+                } else {
+                    panic!("Not f32 or f64");
+                }
+            }
+            Recipe::Butterfly16 => {
+                if id_t == id_f32 {
+                    Arc::new(NeonF32Butterfly16::new(direction)) as Arc<dyn Fft<T>>
+                } else if id_t == id_f64 {
+                    Arc::new(NeonF64Butterfly16::new(direction)) as Arc<dyn Fft<T>>
+                } else {
+                    panic!("Not f32 or f64");
```

Comment:
How would you feel about doing this panic once at the start of this method? It adds quite a lot to the line count of the method as-is.